template-engine/chapter.md

```diff
@@ -377,14 +377,14 @@ append_result("hello")
 
 In the template engine code, we've split it out this way so that we only do the
 first step once, no matter how many times we do the second step.  This saves us
-a small amount of time, because we avoid the time to look up the append attribute.
+a small amount of time, because we avoid taking the time to look up the append attribute.
 
 This is an example of a micro-optimization: an unusual coding technique that
-gains us tiny improvements in speed.  They can be less readable, or more
+gains us tiny improvements in speed.  Micro-optimizations can be less readable, or more
 confusing, so they are only justified for code that is a proven performance
 bottleneck.  Developers disagree on how much micro-optimization is justified,
 and some beginners overdo it.  The optimizations here were added only after
-timing experiments showed that they improved the speed, even if only a little
+timing experiments showed that they improved performance, even if only a little
```

Comment:
Not sure if this retains the same meaning as your original text, but "improved the speed" seems to require more information -- improved the speed of what? Whereas "performance" stands alone.